examples/undocumented/python/graphical/eigenfaces.py

```diff
@@ -92,18 +89,18 @@ def predict(self, image):
         imageAsRow = np.asarray(image.reshape(image.shape[0]*image.shape[1],1),
                                 np.float64);
         #project inthe subspace
-        p = self.pca.apply_to_feature_vector(RealFeatures(imageAsRow).get_feature_vector(0));
+        p = self.pca.apply_to_feature_vector(sg.RealFeatures(imageAsRow).get_feature_vector(0));
 
         #min value to find the face
         minDist =1e100;
         #class
         minClass = -1;
         #search which face is the best match
         for sampleIdx in range(len(self._projections)):
-            test = RealFeatures(np.asmatrix(p,np.float64).T)
-            projection = RealFeatures(np.asmatrix(self._projections[sampleIdx],
+            test = sg.RealFeatures(np.asmatrix(p,np.float64).T)
+            projection = sg.RealFeatures(np.asmatrix(self._projections[sampleIdx],
                                         np.float64).T)
-            dist = EuclideanDistance( test, projection).distance(0,0)
+            dist = sg.EuclideanDistance( test, projection).distance(0,0)
```

Comment:
same here, need to replace with factory

Reply:
which factory?

Reply:
and where can i read about sg factories

Reply:
sg.create_distance("EuclideanDistance")

Reply:
where will test and projection go?

Reply:
`sg.create_distance("EuclideanDistance", lhs=test , rhs=projection)`, you can find those from shogun example btw.

Reply:
where to see shogun examples?

Reply:
thanks btw